scikit-learn fails to build with MKL

**Describe the bug**
scikit-learn fails to build when `blas = mkl` is set for numpy.

**To Reproduce**
Steps to reproduce the behavior:
```
>  nix-shell -p 'python3.withPackages(ps: with ps; [ scikitlearn ])'
[... success]
>  nix-shell -p '(python37.override {packageOverrides = self: super: { numpy = super.numpy.override { blas = mkl;}; }; } ).withPackages(ps: with ps; [ scikitlearn ])'
[...]
=================================== FAILURES ===================================
_________________________________ test_omp_cv __________________________________

    def test_omp_cv():
        y_ = y[:, 0]
        gamma_ = gamma[:, 0]
        ompcv = OrthogonalMatchingPursuitCV(normalize=True, fit_intercept=False,
                                            max_iter=10, cv=5)
        ompcv.fit(X, y_)
>       assert_equal(ompcv.n_nonzero_coefs_, n_nonzero_coefs)

/nix/store/7sz7kr8szqhmy75mx2hxbfz3wvgq2iyx-python3.7-scikit-learn-0.21.2/lib/python3.7/site-packages/sklearn/linear_model/tests/test_omp.py:208:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/nix/store/4g2ilwnk4mj4g68rc52kygx7gcqxnjd0-python3-3.7.5/lib/python3.7/unittest/case.py:852: in assertEqual
    assertion_func(first, second, msg=msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sklearn.utils._unittest_backport.TestCase testMethod=__init__>
first = 6, second = 5, msg = '6 != 5'

    def _baseAssertEqual(self, first, second, msg=None):
        """The default assertEqual implementation, not type specific."""
        if not first == second:
            standardMsg = '%s != %s' % _common_shorten_repr(first, second)
            msg = self._formatMessage(msg, standardMsg)
>           raise self.failureException(msg)
E           AssertionError: 6 != 5

/nix/store/4g2ilwnk4mj4g68rc52kygx7gcqxnjd0-python3-3.7.5/lib/python3.7/unittest/case.py:845: AssertionError
```

full output: https://gist.github.com/tbenst/f32351229c0330edf3aefa003e6c1053

--- a/pkgs/development/python-modules/scikitlearn/default.nix
+++ b/pkgs/development/python-modules/scikitlearn/default.nix
@@ -12,13 +12,13 @@
 
 buildPythonPackage rec {
   pname = "scikit-learn";
-  version = "0.21.2";
+  version = "0.21.3";
   # UnboundLocalError: local variable 'message' referenced before assignment
   disabled = stdenv.isi686;  # https://github.com/scikit-learn/scikit-learn/issues/5534
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1nvj9j16y1hz9gm0qwzpnx2zmz55c63k1fai643migsyll9c7bqa";
+    sha256 = "eb9b8ebf59eddd8b96366428238ab27d05a19e89c5516ce294abc35cea75d003";
   };
 
   buildInputs = [

